bot/exts/utils/clean.py

```diff
@@ -89,6 +163,38 @@ def predicate_regex(message: Message) -> bool:
             await ctx.send(embed=embed)
             return
 
+        if after_message:
+
+            # Ensure that until_message is specified.
+            if not until_message:
+                embed = Embed(
```

Comment:
It seems to me that this embed and others would be better as thrown exceptions, such as `MissingRequiredArgument`, `BadArgument`, or possibly custom errors subclassed from `UserInputError` in `discord.ext.commands.error`.

Reply:
Main reason as to why this logic even exists is to ensure that, if someone decides to use the function, not from the command interface, they'd get a response, am not sure how useful it is.
I think for that matter it could be removed.
As far as regular user is concerned, that should already be the case as per [Line 387](https://github.com/python-discord/bot/blob/cleanrework/bot/exts/utils/clean.py#L387) the command will error out

Reply:
I think this part should be exported to a helper function and replaced with errors. The error handler should translate them into embeds and send them.

Reply:
But yes, if we don't expect the function to be used in another context then I'm not sure there's a point

Reply:
Added in https://github.com/python-discord/bot/pull/1527/commits/ed352272a67224182178bbd5583746053ec912a6